core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/FastDoubleParserTests.kt

```diff
@@ -0,0 +1,163 @@
+package org.jetbrains.kotlinx.dataframe.io
+
+import io.kotest.matchers.collections.shouldContainInOrder
+import org.jetbrains.kotlinx.dataframe.api.ParserOptions
+import org.jetbrains.kotlinx.dataframe.impl.io.FastDoubleParser
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.Locale
+
+private const val LOG_LEVEL = "org.slf4j.simpleLogger.defaultLogLevel"
+
+class FastDoubleParserTests {
+
+    private var loggerBefore: String? = null
+
+    @Before
+    fun setLogger() {
+        loggerBefore = System.getProperty(LOG_LEVEL)
+        System.setProperty(LOG_LEVEL, "debug")
+    }
+
+    @After
+    fun restoreLogger() {
+        if (loggerBefore != null) {
+            System.setProperty(LOG_LEVEL, loggerBefore)
+        }
+    }
+
+    @Test
+    fun `can fast parse doubles`() {
+        val parser = FastDoubleParser(ParserOptions(locale = Locale.ROOT, useFastDoubleParser = true))
+
+        val numbers = listOf(
+            "+12.45",
+            "-13.35",
+            "100123.35",
+            "-204,235.23",
+            "1.234e3",
+            "3e-04", // failed with old double parser
```

Comment:
Will it fail, if we disable fast parsing? Or they aligned now in the terms of supported functionality, locales?
And one just is faster than another?

Reply:
nope, it will fail in the old one. We cannot make the old one behave the same, as it's not supported by NumberFormat.